spatialpandas/geometry/base.py

```diff
@@ -624,13 +739,339 @@ def fillna(self, value=None, method=None, limit=None):
 
     # Base geometry methods
     @property
-    def bounds(self):
-        return bounds_interleaved(self.flat_values)
+    def total_bounds(self):
+        return total_bounds_interleaved(self.flat_values)
+
+    @property
+    def total_bounds_x(self):
+        return total_bounds_interleaved_1d(self.flat_values, 0)
 
     @property
-    def bounds_x(self):
-        return bounds_interleaved_1d(self.flat_values, 0)
+    def total_bounds_y(self):
+        return total_bounds_interleaved_1d(self.flat_values, 1)
 
     @property
-    def bounds_y(self):
-        return bounds_interleaved_1d(self.flat_values, 1)
+    def bounds(self):
+        return bounds_interleaved(self.flat_values, self.flat_outer_offsets)
+
+    def intersects_bounds(self, bounds, inds=None):
+        """
+        Test whether each element in the array intersects with the supplied bounds
+
+        Args:
+            bounds: Tuple of bounds coordinates of the form (x0, y0, x1, y1)
+            inds: Optional array of indices into the array. If supplied, intersection
+                calculations will be performed only on the elements selected by this
+                array.  If not supplied, intersection calculations are performed
+                on all elements.
+
+        Returns:
+            Array of boolean values indicating which elements of the array intersect
+            with the supplied bounds
+        """
+        raise NotImplementedError()
+
+
+class _CoordinateIndexer(object):
+    def __init__(self, obj, parent=None):
+        self._obj = obj
+        self._parent = parent
+
+    def __getitem__(self, key):
+        obj = self._obj
+        xs, ys = key
+        # Handle xs and ys as scalar numeric values
+        if type(xs) is not slice:
+            xs = slice(xs, xs)
+        if type(ys) is not slice:
+            ys = slice(ys, ys)
+
+        if xs.step is not None or ys.step is not None:
+            raise ValueError("step not supported")
```

Comment:
Can you explain a bit?  E.g. "step not supported in cases __ and ___; instead use ___"?

Reply:
FWIW, this is the logic `Geopandas.cx` uses.  I think the core issue is that `cx` uses slices as a way to represent intervals in continuous spaces (rather than as a specification of a range of discrete values).  Under this interpretation, I don't think a step really has a clear meaning.

Reply:
Attempt made in 8bcd3a0

Reply:
Thanks!